src/GitUtils.ps1

```diff
@@ -384,41 +388,41 @@ function Get-GitStatus {
             }
         }
 
-        if (!$branch) { $branch = Get-GitBranch $GitDir $sw }
+        $branch = Get-GitBranch -Branch $branch -GitDir $GitDir -IsDotGitOrBare:$isDotGitOrBare -sw $sw
```

Comment:
This is a behavior change when `$branch` is truthy but you're the Git expert.  :-)

Reply:
> This is a behavior change when `$branch` is truthy but you're the Git expert. :-)

Yep, that's the point. We've been missing some suffixes because we haven't been calling `Get-GitBranch` if we already know the branch. It's been working for rebase because it detaches `HEAD`.